DoSomething/ds-mdata-responder
# Player names accepted and stored from game creation

#### What's this PR do?
Takes and stores beta player names provided during game creation to the user and game docs.

#### Where should the reviewer start?
##### Gameplay
Beta names are expected in `beta_first_name_0`, `beta_first_name_1`, and `beta_first_name_2`. Line 98 of SGCompetitiveStoryController.js takes those values and stores it in the game doc being created. Below that, `createPlayer()` and its use have been modified to store the names in the user docs as well, as long as they values are not null.

##### Testing
I took the `before` out of `testHelperFunctions.gameAppSetup`. When scanning mocha tests I found it a little confusing what gameAppSetup exactly accomplished. By moving the responsibility of declaring the `before` to each of the tests, I think it makes the flow of a test a bit clear, and keeps the gameAppSetup code independent of always needing to be used at the start of a test. (Though I admittedly don't know when else one would want to use it.)

And then in test/GameAlphaStart.js, I added a test for verifying data after a game creation with names. It's basically a copy of the test above it, but with the names and document validation.

#### How should this be manually tested?
* npm test
* Create a game through Postman, and include beta_first_name_0, beta_first_name_1, and beta_first_name_2 in the request. Then check in your database that the name is saved in both the game doc and user docs.

#### What are the relevant tickets?
Closes #328
